doc/api/connection.md

```diff
@@ -0,0 +1,158 @@
+# Connection
+
+## Class: Connection
+
+### new Connection(transport, server, client)
+
+* transport: `Transport`.
+* server: `Server` — JSTP server instance, used only for server—side parts
+  of connections (optional, but either server or client is required).
+* client: `Client` — JSTP client instance, used only for client—side parts
+  of connections (optional, but either server or client is required).
+
+Don't call this constructor manually unless you use custom tranport.
+Recommended approach is to call `connect()` function provided by these modules:
+* `net`
+* `tls`
+* `ws`
+* `wss`
+
+### Methods
+
+#### connection.handshake(app, \[login\], \[password\], callback)
+
+* app: `String || Object` — application to connect to as `'name'` or
+  `'name@version'` or `{ name, version }`, where version must be
+  a valid semver range.
+* login: `String` — user name.
+* password: `String` — user password.
+* callback(error, sessionId) — callback function to invoke after the handshake
+  is completed.
+  * error: `Error`.
+  * sessionId: `number`.
```

Comment:
Ain't it a string? [here](https://github.com/metarhia/jstp/blob/master/lib/simple-auth-policy.js#L30)